src/clinical/donor-repo.ts

```diff
@@ -109,20 +110,22 @@ export interface DonorRepository {
 	): Promise<DeepReadonly<Donor> | undefined>;
 	iterateAllByProgramId(programId: string): AsyncIterable<DeepReadonly<Donor>>;
 	create(donor: DeepReadonly<Partial<Donor>>): Promise<DeepReadonly<Donor>>;
-	update(donor: DeepReadonly<Donor>): Promise<DeepReadonly<Donor>>;
-	updateAll(donors: DeepReadonly<Donor>[]): Promise<DeepReadonly<Donor>[]>;
+	update(donor: Donor): Promise<DeepReadonly<Donor>>;
+	updateAll(donors: Donor[]): Promise<DeepReadonly<Donor>[]>;
 	countBy(filter: any): Promise<number>;
 }
 
 // Mongoose implementation of the DonorRepository
 export const donorDao: DonorRepository = {
 	async insertDonors(donors: Donor[]) {
-		await mongoose.connection.db.collection('donors').insertMany(donors);
+		const donorsWithIds = await setEntityIdsForDonors(donors);
+		await mongoose.connection.db.collection('donors').insertMany(donorsWithIds);
 	},
 	async updateDonor(donor: Donor) {
+		const donorsWithIds = await setEntityIds(donor);
```

Comment:
Both `setEntityIds` functions throw errors when network errors occur, should there be some sort of error handling here to handle failures or is this now expected to be handled by the caller of `updateDonor` and `insertDonors`?